client/batch_controller.go

```diff
@@ -89,50 +94,49 @@ func (tbc *tsoBatchController) fetchPendingRequests(ctx context.Context, tsoRequ
 		}
 
 		// The token is ready. If the first request didn't arrive, wait for it.
-		if tbc.collectedRequestCount == 0 {
+		if bc.collectedRequestCount == 0 {
 			select {
 			case <-ctx.Done():
 				return ctx.Err()
-			case firstRequest := <-tsoRequestCh:
-				tbc.pushRequest(firstRequest)
+			case firstRequest := <-requestCh:
+				bc.pushRequest(firstRequest)
 			}
 		}
 
 		// Both token and the first request have arrived.
 		break
 	}
 
-	tbc.extraBatchingStartTime = time.Now()
+	bc.extraBatchingStartTime = time.Now()
 
-	// This loop is for trying best to collect more requests, so we use `tbc.maxBatchSize` here.
+	// This loop is for trying best to collect more requests, so we use `bc.maxBatchSize` here.
 fetchPendingRequestsLoop:
-	for tbc.collectedRequestCount < tbc.maxBatchSize {
+	for bc.collectedRequestCount < bc.maxBatchSize {
 		select {
-		case tsoReq := <-tsoRequestCh:
-			tbc.pushRequest(tsoReq)
+		case req := <-requestCh:
+			bc.pushRequest(req)
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
 			break fetchPendingRequestsLoop
 		}
 	}
 
-	// Check whether we should fetch more pending TSO requests from the channel.
-	// TODO: maybe consider the actual load that returns through a TSO response from PD server.
-	if tbc.collectedRequestCount >= tbc.maxBatchSize || maxBatchWaitInterval <= 0 {
+	// Check whether we should fetch more pending requests from the channel.
+	if bc.collectedRequestCount >= bc.maxBatchSize || maxBatchWaitInterval <= 0 {
```

Comment:
Do they share the same `maxBatchWaitInterval`?

Reply:
`maxBatchWaitInterval` is passed in via the `fetchPendingRequests` function so that it can be changed along with the caller, e.g., the TSO batch controller and the region query batch controller.